### PR TITLE
MTDSA-12968 RAML for itemDetails #1553 Retrieve Self Assessment Balance and Transactions

### PR DESCRIPTION
**Pull request by abigail-houston-hmrc**

### CODE REVIEW

File: resources/public/api/conf/2.0/examples/retrieveBalanceAndTransactions/itemDetails.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "subItem": "001",
+  "dueDate": "2018-08-13",
+  "amount": 10000.00,
+  "clearingDate": "2018-08-13",
+  "clearingReason": "01",
+  "outgoingPaymentMethod": "outgoing Payment",
+  "isChargeOnHold": False,
```
Comment (LouiseWong98) [inline resources/public/api/conf/2.0/examples/retrieveBalanceAndTransactions/itemDetails.json:8]: should be false

File: resources/public/api/conf/2.0/schemas/retrieveBalanceAndTransactions/itemDetails.json [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      "description": "When a payment is used against a charge, this links payment and charge together.",
+      "type": "string",
+      "example": "3350000253"
+    },
+    "isChargeEstimate": {
+      "description": "Estimated charges due for the customer",
+      "type": "boolean",
+      "example": "true"
+    },
+    ,
```
Comment (LouiseWong98) [inline resources/public/api/conf/2.0/schemas/retrieveBalanceAndTransactions/itemDetails.json:122]: remove the extra ,

File: resources/public/api/conf/2.0/examples/retrieveBalanceAndTransactions/itemDetails.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "subItem": "001",
+  "dueDate": "2018-08-13",
+  "amount": 10000.00,
+  "clearingDate": "2018-08-13",
+  "clearingReason": "01",
+  "outgoingPaymentMethod": "outgoing Payment",
```
Comment (LouiseWong98) [inline resources/public/api/conf/2.0/examples/retrieveBalanceAndTransactions/itemDetails.json:7]: should be "Payable Order Repayment"

File: resources/public/api/conf/2.0/examples/retrieveBalanceAndTransactions/itemDetails.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (LouiseWong98) [inline resources/public/api/conf/2.0/examples/retrieveBalanceAndTransactions/itemDetails.json:1]: would it be easier to have items in it as well?

File: resources/public/api/conf/2.0/schemas/retrieveBalanceAndTransactions/itemDetails.json [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+      "example": "2018-04-05"
+    },
+    "clearingReason": {
+      "description": "The reason for the change to the item.",
+      "type": "string",
+      "example": "01"
+    },
+    "outgoingPaymentMethod": {
+      "description": "The method used to make the outgoing payment.",
+      "oneOf": [
```
Comment (LouiseWong98) [inline resources/public/api/conf/2.0/schemas/retrieveBalanceAndTransactions/itemDetails.json:37]: would be better to add these in the description and remove the oneOf array

File: resources/public/api/conf/2.0/schemas/retrieveBalanceAndTransactions/itemDetails.json [hunk 6]
```diff
@@ -0,0 +1,130 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Retrieve History of a Self Assessment Charge",
+  "description": "This endpoint enables you to retrieve the history of changes to a single charge made by HMRC. A National Insurance number and transaction ID must be provided.",
+  "type": "object",
+  "properties": {
```
Comment (LouiseWong98) [inline resources/public/api/conf/2.0/schemas/retrieveBalanceAndTransactions/itemDetails.json:6]: an extra items object is missing

File: resources/public/api/conf/2.0/examples/retrieveBalanceAndTransactions/itemDetails.json (hunk 4 above)
Comment (abigail-houston-hmrc) [reply resources/public/api/conf/2.0/examples/retrieveBalanceAndTransactions/itemDetails.json:1]: Will probably need to merge all the examples into one file, so will update after next commit

File: resources/public/api/conf/2.0/schemas/retrieveBalanceAndTransactions/itemDetails.json [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+            "description": "When a payment is used against a charge, this links payment and charge together.",
+            "type": "string",
+            "example": "3350000253"
+          },
+          "isChargeEstimate": {
+            "description": "Estimated charges due for the customer",
+            "type": "boolean",
+            "example": "true"
+          },
+          "required": [
```
Comment (LouiseWong98) [inline resources/public/api/conf/2.0/schemas/retrieveBalanceAndTransactions/itemDetails.json:109]: This is in the wrong place, it should be placed after the next bracket - in line 115

File: resources/public/api/conf/2.0/schemas/retrieveBalanceAndTransactions/itemDetails.json [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+            "example": "2018-04-05"
+          },
+          "clearingReason": {
+            "description": "The reason for the change to the item.",
+            "type": "string",
+            "example": "01"
+          },
+          "outgoingPaymentMethod": {
+            "description": "The method used to make the outgoing payment. One of Repayment to Card, Payable Order Repayment or BACS Payment out",
+            "example": "Payable Order Repayment"
```
Comment (LouiseWong98) [inline resources/public/api/conf/2.0/schemas/retrieveBalanceAndTransactions/itemDetails.json:44]: "enum" :[
              "Repayment to Card",
              "Payable Order Repayment",
              "BACS Payment out"
            ]
is missing

**Pull request merged**